src/main/resources/assets/js/components/Marathon.jsx

```diff
@@ -36,11 +42,57 @@ var Marathon = React.createClass({
       deploymentsFetchState: States.STATE_LOADING,
       fetchState: States.STATE_LOADING,
       modalClass: null,
+      route: null,
       tasksFetchState: States.STATE_LOADING
     };
   },
 
   componentDidMount: function () {
+    var router = this.props.router;
+
+    router.on("route", function (route, params) {
+      this.setState({
+        route: {
+          name: route,
+          params: params
+        }
+      });
+    }.bind(this));
+
+    router.on("route:about", function () {
+      this.modalDestroy();
+      this.setState({
+        modalClass: AboutModalComponent
+      });
+    }.bind(this));
+
+    router.on("route:apps", function (appid) {
+      if (appid) {
+        if (this.state.activeAppId !== appid) {
+          this.modalDestroy();
+        }
+
+        this.setState({
+          activeAppId: appid,
+          activeApp: this.state.collection.get("/" + appid),
```

Comment:
Yeah, need both, because activeAppId is used here:
https://github.com/mesosphere/marathon/pull/1066/files#diff-6876cc8f56c80c7276969fd16e24b309R180
On first page load the collection isn't ready, so this is undefined:
https://github.com/mesosphere/marathon/pull/1066#discussion-diff-23328110R77